lib/Twig/Parser.php

```diff
@@ -114,7 +114,7 @@ public function parse(Twig_TokenStream $stream, $test = null, $dropNeedle = fals
             throw $e;
         }
 
-        $node = new Twig_Node_Module(new Twig_Node_Body(array($body)), $this->parent, new Twig_Node($this->blocks), new Twig_Node($this->macros), new Twig_Node($this->traits), $this->embeddedTemplates, $this->getFilename(), $stream->getSource());
+        $node = new Twig_Node_Module(new Twig_Node_Body(array($body)), $this->parent, new Twig_Node($this->blocks), new Twig_Node($this->macros), new Twig_Node($this->traits), $this->embeddedTemplates, $this->getFilename(), $this->env->isDebug() ? $stream->getSource() : '');
```

Comment:
I would not do the check here. I would always pass the source to the node, and check for the debug mode when compiling the node